cv_next/app/actions/users/handleUserState.ts

```diff
@@ -0,0 +1,29 @@
+"use server";
+
+import SupabaseHelper from "@/server/api/supabaseHelper";
+import { ProfileKeys } from "@/lib/supabase-definitions";
+import { getUserModel } from "./getUser";
+
+/**
+ * Handle which user can enter a given route(used instead of the middleware to reduce server requests)
+ * If there's no user connect, return login and redirect there
+ * If the user is inactive, return inactive and redirect there
+ * If the user is active, return the given finalRedirect and redirect there
+ * @param {string} finalRedirect - The path the user wants to access
+ * @return {Promise<string>} a promised string with where to redirect the user to
+ */
+export const handleCurrentUser = async (
```

Comment:
What does "handle" mean? Please rename this (and the file). Do you mean e.g. `userAccessValidation`, or something lik that?

Reply:
Also I see the usage of this, and I think this is a good example of bad separation of concerns. The function should return the validation, not what to do with it. The return values should be e.g. 0 = allowed, 1 = not logged in, 2 = not active (can be semantic-ized with enums e.g. AccessType.Allowed = 0).
The function that uses this method should decide based on the return value what to do (e.g. which redirection to do).

Reply:
Now that we managed to solve the problem with the SSR and the login in #72 we can actually remove this as the middleware handles this